### ISSUE TITLE
Bad comparison for 'typeof'

### ISSUE THREAD

**Issue by unional** [palantir/tslint#3508]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.6.1
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
if (typeof a !== typeof b)
  throw new Error('Type mismatch')
```

#### Actual behavior
Error: Bad comparison for 'typeof'

#### Expected behavior
I think this should be excluded as bad comparison.
